components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/utils/SchemaValidationUtils.java

```diff
@@ -0,0 +1,133 @@
+
+        if (name != null && map.containsKey(name)) {
+
+            return map.get(name).stream()
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
+        } else {
+            return Collections.emptyList();
+        }
+
+    }
+
+    /**
+     * Method to remove query params from request path.
+     *
+     * @param subResourcePath full request path.
+     * @return cleaned request path.
+     */
+    public static String getRestSubRequestPath(String subResourcePath) {
+
+        if (subResourcePath.contains("?")) {
+            return subResourcePath.split("\\?")[0];
+        } else if (subResourcePath.contains("#")) {
+            return subResourcePath.split("#")[0];
+        } else {
+            return subResourcePath;
+        }
+    }
+
+    /**
+     * Utility function to extract query params from resource path.
+     *
+     * @param apiResource reource path with query params.
+     * @param path        path without query params.
+     * @return Map of Query params.
+     * @throws UnsupportedEncodingException When failed to decode
+     */
+    public static Map<String, Collection<String>> getQueryParams(String apiResource, String path)
+            throws UnsupportedEncodingException {
+
+        Map<String, String> queryParams = new HashMap<>();
+        if (!apiResource.equals(path) && apiResource.contains("?")) {
+            String queryString = apiResource.replace(path + "?", "");
+            String[] query = queryString.split("&");
+            for (String keyValue : query) {
+                int idx = keyValue.indexOf("=");
+                queryParams.put(
+                        URLDecoder.decode(keyValue.substring(0, idx), "UTF-8"),
+                        URLDecoder.decode(keyValue.substring(idx + 1), "UTF-8"));
+            }
+        }
+        return queryParams.entrySet()
+                .stream()
+                .collect(Collectors.toMap(
+                        Map.Entry::getKey, entry -> Collections.singleton(entry.getValue())));
+    }
+
+    /**
+     * Get the Request/Response messageContent as a JsonObject.
+     *
+     * @param messageContext Message context
+     * @return JsonElement which contains the request/response message content
+     */
+    public static Optional<String> getMessageContent(MessageContext messageContext) {
+
+        Optional<String> payloadObject = Optional.empty();
+        org.apache.axis2.context.MessageContext axis2Context = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        if (JsonUtil.hasAJsonPayload(axis2Context)) {
+            payloadObject = Optional.of(JsonUtil.jsonPayloadToString(axis2Context));
+        } else if (messageContext.getEnvelope().getBody() != null) {
+            Object objFirstElement = messageContext.getEnvelope().getBody().getFirstElement();
```

Comment:
I believe need to do a  null check here. WDYT?
messageContext.getEnvelope().getBody()

Reply:
no need you already have done

Reply:
I have done that null check in line 119